_studio/mfx_lib/decode/mpeg2/src/mfx_mpeg2_decode.cpp

```diff
@@ -651,31 +651,36 @@ mfxStatus VideoDECODEMPEG2::DecodeFrameCheck(mfxBitstream* bs, mfxFrameSurface1*
     MFX_CHECK(m_core, MFX_ERR_UNDEFINED_BEHAVIOR);
     MFX_CHECK(m_decoder, MFX_ERR_NOT_INITIALIZED);
 
-    MFX_CHECK_NULL_PTR2(surface_work, surface_out);
+    MFX_CHECK_NULL_PTR1(surface_out);
 
     mfxStatus sts = bs ? CheckBitstream(bs) : MFX_ERR_NONE;
     MFX_CHECK_STS(sts);
 
-    if (m_opaque)
+    if (surface_work != nullptr)
     {
-        sts = CheckFrameInfoCodecs(&surface_work->Info, MFX_CODEC_MPEG2);
-        MFX_CHECK(MFX_ERR_NONE == sts, MFX_ERR_UNSUPPORTED);
+        if (m_opaque)
+        {
+            sts = CheckFrameInfoCodecs(&surface_work->Info, MFX_CODEC_MPEG2);
+            MFX_CHECK(MFX_ERR_NONE == sts, MFX_ERR_UNSUPPORTED);
 
-        if (surface_work->Data.MemId || surface_work->Data.Y || surface_work->Data.R || surface_work->Data.A || surface_work->Data.UV) // opaq surface
-            return MFX_ERR_UNDEFINED_BEHAVIOR;
+            if (surface_work->Data.MemId || surface_work->Data.Y || surface_work->Data.R || surface_work->Data.A || surface_work->Data.UV) // opaq surface
```

Comment:
I would suggest to use ```MFX_CHECK``` macro, it will allow to trace triggering of this condition